File: clients/typescript/examples/admin-find-reranking-providers.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

const client = new DataAPIClient("**APPLICATION_TOKEN**");

const admin = client.admin();

const databaseAdmin = admin.dbAdmin("**API_ENDPOINT**");

(async function () {
  const providers = await databaseAdmin.findRerankingProviders();

  console.log(JSON.stringify(providers));
})();

File: clients/java/examples/admin-find-reranking-providers.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.admin.DatabaseAdmin;
import com.datastax.astra.client.core.rerank.RerankProvider;
import com.datastax.astra.client.databases.Database;
import com.datastax.astra.client.databases.commands.results.FindRerankingProvidersResult;
import java.util.Map;

public class Example {
  public static void main(String[] args) {
    Database database = new DataAPIClient("**APPLICATION_TOKEN**").getDatabase("**API_ENDPOINT**");

    DatabaseAdmin databaseAdmin = database.getDatabaseAdmin();

    FindRerankingProvidersResult result = databaseAdmin.findRerankingProviders();

    Map<String, RerankProvider> providers = result.getRerankingProviders();

    System.out.println(providers);
  }
}
